python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/perf/partition_mappings_galore_perf_scenario.py

```diff
@@ -0,0 +1,96 @@
+from typing import Optional, Sequence
+
+from dagster import (
+    AssetDep,
+    AssetsDefinition,
+    AutoMaterializePolicy,
+    DailyPartitionsDefinition,
+    Definitions,
+    HourlyPartitionsDefinition,
+    PartitionsDefinition,
+    RunRequest,
+    TimeWindowPartitionMapping,
+    asset,
+)
+from dagster._core.storage.tags import (
+    ASSET_PARTITION_RANGE_END_TAG,
+    ASSET_PARTITION_RANGE_START_TAG,
+)
+
+from .perf_scenario import ActivityHistory, PerfScenario
+
+
+def asset_def(
```

Comment:
it does feel like this is fairly parallel to the other recent change in AMP-testing land, which is the `AssetSpecWithPartitionsDef`. Not saying that `AssetSpecWithPartitionsDef` is specifically more elegant than this, but I do think that it'd be nice to converge on a single API for building asset graphs for tests. i.e. I could imagine something like

```python
my_defs = ScenarioState.with_asset(
    "a_1", partitions_def=...
).with_asset(
    "a_2", partitions_def=...
).with_asset(
    "b", deps=..., partitions_def=...
).defs
```

no action required on this PR, as that API doesn't currently exist, but could be nice (asset backfill testing also makes use of this sort of pattern of building asset graphs, and I imagine it's generally useful to have ergonomic ways to describe asset graphs you want to test)

Reply:
Yeah I definitely agree that it would be useful to have a standardized ergonomic way of specifying asset graphs we want to test, and would eagerly refactor this code on top of one.

Getting into specifics, it's not clear to me that the chained-invocations pattern is more ergonomic than just constructing out of a list of asset defs.

Reply:
yeah the main benefit of the chained invocations pattern is that it makes it easy to first define some basic shape (e.g. a diamond) and then later layer on specific properties (partitions definitions, AMPs, etc.). that makes it so you can reuse the same shape multiple times

not particularly helpful in this specific case